plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts

```diff
@@ -130,4 +208,350 @@ export class ArgoService implements ArgoServiceApi {
       throw new Error('Could not retrieve ArgoCD app data.');
     }
   }
+
+  async createArgoProject(
+    baseUrl: string,
+    argoToken: string,
+    projectName: string,
+    namespace: string,
+    sourceRepo: string,
+  ): Promise<object> {
+    const data = {
+      project: {
+        metadata: {
+          name: projectName,
+        },
+        spec: {
+          destinations: [
+            {
+              name: 'local',
+              namespace: namespace,
+              server: 'https://kubernetes.default.svc',
```

Comment:
Would this be something that may need different addresses for different cluster configurations or is it always the same?

Reply:
In our case its always the same however I see your point. I have made this an optional input parameter with a default value.